core/frontend/src/IModelConnection.ts

```diff
@@ -345,15 +345,15 @@ export abstract class IModelConnection extends IModel {
     return undefined;
   }
 
-  /** Request element mass properties from the backend.
-   * @note For better performance use [[getMassPropertiesPerCandidate]] when called from a loop with identical operations and a single candidate per iteration.
-   */
+  /** Request element mass properties from the backend. */
   public async getMassProperties(requestProps: MassPropertiesRequestProps): Promise<MassPropertiesResponseProps> {
     return IModelReadRpcInterface.getClientForRouting(this.routingContext.token).getMassProperties(this.getRpcProps(), requestProps);
   }
 
-  /** Request mass properties for multiple elements from the backend. */
-  public async getMassPropertiesPerCandidate(requestProps: MassPropertiesPerCandidateRequestProps): Promise<MassPropertiesPerCandidateResponseProps[]> {
+  /** Request mass properties for multiple elements from the backend.
+   * @deprecated in 4.11. Use [[IModelConnection.getMassProperties]].
```

Comment:
Is it possible to replicate this functionality using `getMassProperties`?

Reply:
Definitely, if you don't mind the network traffic / latency that it adds to call that method over and over again to get mass properties for all of the elements that you need.

`getMassPropertiesPerCandidate` calculates mass properties separately for every candidate.
`getMassProperties` calculates the sum of mass properties of all provided candidates.